L1Trigger/TrackFindingTracklet/interface/L1TStub.h

```diff
@@ -0,0 +1,133 @@
+#ifndef L1Trigger_TrackFindingTracklet_interface_L1TStub_h
+#define L1Trigger_TrackFindingTracklet_interface_L1TStub_h
+
+#include <iostream>
+#include <fstream>
+#include <cassert>
+#include <cmath>
+#include <vector>
+
+namespace Trklet {
+
+  class L1TStub {
+  public:
+    L1TStub();
+    L1TStub(int eventid,
+            std::vector<int> tps,
+            int iphi,
+            int iz,
+            int layer,
+            int ladder,
+            int module,
+            int strip,
+            double x,
+            double y,
+            double z,
+            double sigmax,
+            double sigmaz,
+            double pt,
+            double bend,
+            int isPSmodule,
+            int isFlipped);
+
+    void AddInnerDigi(int ladder, int module, int irphi, int iz);
+    void AddOuterDigi(int ladder, int module, int irphi, int iz);
+
+    void write(std::ofstream& out);
+    void write(std::ostream& out);
+
+    int ptsign();
+
+    double diphi();
+
+    double iphiouter();
+
+    double diz();
+
+    unsigned int layer() const { return layer_; }
+    int disk() const {
+      if (z_ < 0.0) {
+        return -module_;
+      }
+      return module_;
+    }
+    unsigned int ladder() const { return ladder_; }
+    unsigned int module() const { return module_; }
+    std::vector<std::pair<int, int> > innerdigis() const { return innerdigis_; }
+    std::vector<std::pair<int, int> > outerdigis() const { return outerdigis_; }
+    std::vector<std::pair<int, int> > innerdigisladdermodule() const { return innerdigisladdermodule_; }
+    std::vector<std::pair<int, int> > outerdigisladdermodule() const { return outerdigisladdermodule_; }
+    double x() const { return x_; }
+    double y() const { return y_; }
+    double z() const { return z_; }
+    double r() const { return sqrt(x_ * x_ + y_ * y_); }
```

Comment:
I think `hypot(x_, y_)` is a faster alternative ?

Reply:
changed as suggested

Reply:
not having been familiar with the `hypot` function I looked online and [here](https://stackoverflow.com/questions/32435796/when-to-use-stdhypotx-y-over-stdsqrtxx-yy) it is suggested that while `hypot` is safer with regards to over- or underflow in intermediate computation steps, people find it be slower than the `std::sqrt` implementation. If the range of values that will go into the calculation are unlikely (impossible?) to cause overflow, one could consider staying with/reverting to the previous version